### PR TITLE
Add remaining docs-rs-staging infra except for ecs app

### PR DESCRIPTION
**Pull request by rylev**
This adds the remaining docs-rs-staging infra with the exception of the ecs app which I would like to do in a follow-up PR. 

Since we do not have an ecs app configured the cloud front distribution is pointed to example.com as its origin. Once the ecs app is configured, we will change the origin to the ecs app. 

Also, the `aws_s3_bucket_inventory` used for inventory storage of the main s3 bucket is now configurable through a variable. This variable determines whether to do the inventory storage or not. As is currently configured, docs-rs-staging will *not* have a perform this inventory storage.

### CODE REVIEW

File: terragrunt/modules/docs-rs/s3.tf [hunk 1]
```diff
@@ -0,0 +1,106 @@
+resource "aws_s3_bucket" "storage" {
+  bucket = "static-docs-rs"
```
Comment (rylev) [inline terragrunt/modules/docs-rs/s3.tf:2]: Needs to be globally unique. Perhaps post fix with the account id.

File: terragrunt/modules/docs-rs/s3.tf [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+  included_object_versions = "Current"
+  optional_fields          = ["ETag", "ReplicationStatus", "Size"]
+
+  schedule {
+    frequency = "Weekly"
+  }
+  destination {
+    bucket {
+      bucket_arn = var.inventories_bucket_arn
```
Comment (rylev) [inline terragrunt/modules/docs-rs/s3.tf:47]: Just create the bucket. 

Question: does this need to be configurable?

File: terragrunt/accounts/docs-rs-staging/docs-rs/terragrunt.hcl [hunk 3]
```diff
@@ -0,0 +1,23 @@
+terraform {
```
Comment (rylev) [inline terragrunt/accounts/docs-rs-staging/docs-rs/terragrunt.hcl:1]: Ask in Zulip whether we need different states for the web and the builder (considering they share resources like s3 and RDS).

File: terragrunt/modules/docs-rs/s3.tf [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+  bucket = aws_s3_bucket.storage.id
+
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+resource "aws_s3_bucket" "backups" {
+  bucket = "docs.rs-backups"
```
Comment (rylev) [inline terragrunt/modules/docs-rs/s3.tf:64]: Make name globally unique

File: terragrunt/modules/docs-rs/s3.tf [hunk 5]
```diff
@@ -0,0 +1,106 @@
+resource "aws_s3_bucket" "storage" {
```
Comment (rylev) [inline terragrunt/modules/docs-rs/s3.tf:1]: Change this name to "static-storage"

File: terragrunt/accounts/docs-rs-staging/docs-rs/terragrunt.hcl (hunk 3 above)
Comment (rylev) [reply terragrunt/accounts/docs-rs-staging/docs-rs/terragrunt.hcl:1]: It was agreed in Zulip that this does not need to happen. We should keep them in the same state for simplicity sake.

**Review by jdno — APPROVED** [review]
Thanks, @ryan!

**Comment by rylev** [comment]
@jdno I had to push a new commit to get things work. This is now all applied to the docs-rs-staging account. Please have a look and let me know if I need to adjust in a new PR.

**Pull request merged**